plugin/build_baron.go

```diff
@@ -127,10 +131,9 @@ func IsWebhookConfigured(project string, version string) (evergreen.WebHook, boo
 	if flags.PluginAdminPageDisabled {
 		if version == "" {
 			lastGoodVersion, err := model.FindVersionByLastKnownGoodConfig(project, -1)
-			if err != nil || lastGoodVersion == nil {
-				return evergreen.WebHook{}, false
+			if err == nil && lastGoodVersion != nil {
```

Comment:
Now if version == "" even after checking lastGoodVersion, we make an unnecessary parserProject DB query. I think you want to add an `if version != ""` and rearrange the ifs/elses